### PR TITLE
Add partial update support.

### PR DESCRIPTION
**Pull request by braudabaugh**
Pull Request for issue #128

### CODE REVIEW

**Review by Igorbek — CHANGES_REQUESTED** [review]
~Please see my objections here: https://github.com/graphql-dotnet/conventions/issues/128#issuecomment-462521200~ (outdated)

**Comment by tlil** [comment]
@braudabaugh can you add some unit tests, please? Also cover when `Optional<>` is attempted used for non-nullable types to prove that it handles that correctly.

**Comment by tlil** [comment]
> Please see my objections here: [#128 (comment)](https://github.com/graphql-dotnet/conventions/issues/128#issuecomment-462521200)

@Igorbek You're fine with this now I take it? (from reading your further comments)

**Comment by tlil** [comment]
On another note, I'm looking at better handling of this and nullable/non-nullable types in general for vNext

**Review by tlil — CHANGES_REQUESTED** [review]
See above comments. Thanks for this, btw

**Comment by braudabaugh** [comment]
I've added the unit tests that you asked for.  If you can merge this that would be great.

File: src/GraphQL.Conventions/Types/Optional.cs [hunk 1]
```diff
@@ -30,14 +30,19 @@ public Optional(T value, bool isSpecified)
...
+        { 
             var typeInfo = typeof(T).GetTypeInfo();
             if ((typeInfo.IsValueType && !typeInfo.IsGenericType(typeof(Nullable<>))) || typeInfo.IsGenericType(typeof(NonNull<>)))
             {
                 throw new InvalidOperationException(
                     string.Format("Cannot instantiate with non-nullable type: {0}",
                         typeof(T)));
             }
-        }
+        }
```
Comment (tlil) [inline src/GraphQL.Conventions/Types/Optional.cs]: ? :-) #nitpick

**Review by tlil — COMMENTED** [review]
Thanks for adding the unit tests

**Review by tlil — APPROVED** [review]

**Comment by mgkeen** [comment]
Thanks for doing this! This is exactly what I needed. It doesn't look as though it's included in the latest release though. Any chance a new release can be made with this in?

**Comment by tlil** [comment]
> Thanks for doing this! This is exactly what I needed. It doesn't look as though it's included in the latest release though. Any chance a new release can be made with this in?

Yep, it's on it's way. I've just been out with a back injury for the past 7-8 weeks, so been a bit slow sorry. Keep an eye out over the next few days. :-)

**Comment by agustin-giacchello** [comment]
Hi @tlil.

I hope you are doing better with that injury. I am newbie with GraphQL dotnet and I was wondering, this is the only way to support partial updates with .net and GraphQL?

Regards,
Agus.

**Comment by mgkeen** [comment]
@tlil sorry to hear about your injury. Hope it's healing well!

@agustin-giacchello i'm not using the conventions package yet, so wrote something to do it with the base library. I based it off the discussion here https://github.com/graphql-dotnet/graphql-dotnet/issues/541 . It's essentially along the same lines as this PR. 

It deserializes into a `Dictionary<string,object>`, then uses reflection to assign the fields that are present in the update to an object with `Option<>` properties on them.

If i had known about this conventions package when I started writing my code though, I would used it!

**Comment by tlil** [comment]
> Thanks for doing this! This is exactly what I needed. It doesn't look as though it's included in the latest release though. Any chance a new release can be made with this in?

Now available in 2.1.0 @mgkeen - apologies for the delay.

**Pull request merged**